Sources/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -367,3 +376,130 @@ private extension StoreProduct {
     }
 
 }
+
+/// Implementation of StoreProductType for web billing products
+private final class WebBillingStoreProduct: StoreProductType {
+    let product: WebProductsResponse.Product
+
+    init(product: WebProductsResponse.Product) {
+        self.product = product
+    }
+
+    var productCategory: StoreProduct.ProductCategory {
+        switch self.product.productType {
+        case "subscription":
+            return .subscription
+        default:
+            return .nonSubscription
+        }
+    }
+
+    var productType: StoreProduct.ProductType {
+        switch self.product.productType {
+        case "subscription":
+            return .autoRenewableSubscription
+        case "consumable":
+            return .consumable
+        case "non_consumable":
+            return .nonConsumable
+        default:
+            Logger.error("Unknown web product type: \(self.product.productType)")
+            return .autoRenewableSubscription
+        }
+    }
+
+    var localizedDescription: String {
+        return self.product.description ?? ""
+    }
+
+    var localizedTitle: String {
+        return self.product.title
+    }
+
+    var currencyCode: String? {
+        return self.defaultPurchaseOption?.basePrice?.currency ??
+            self.defaultPurchaseOption?.base?.price.currency
+    }
+
+    var price: Decimal {
+        if let basePrice = self.defaultPurchaseOption?.basePrice {
+            return Decimal(basePrice.amountMicros) / 1_000_000
+        } else if let base = self.defaultPurchaseOption?.base {
+            return Decimal(base.price.amountMicros) / 1_000_000
+        }
+        return 0
+    }
+
+    var localizedPriceString: String {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        if let currencyCode = self.currencyCode {
+            formatter.currencyCode = currencyCode
+        }
+        return formatter.string(from: self.price as NSDecimalNumber) ?? "\(self.price)"
+    }
+
+    var productIdentifier: String {
+        return self.product.identifier
+    }
+
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+    var isFamilyShareable: Bool {
+        return false
+    }
+
+    var subscriptionGroupIdentifier: String? {
+        return nil
+    }
+
+    var priceFormatter: NumberFormatter? {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        if let currencyCode = self.currencyCode {
+            formatter.currencyCode = currencyCode
+        }
+        return formatter
+    }
+
+    var subscriptionPeriod: SubscriptionPeriod? {
+        guard let base = self.defaultPurchaseOption?.base,
+              let periodDuration = base.periodDuration else {
+            return nil
+        }
+
+        // Parse period duration string (e.g. "P1M" for 1 month)
+        // Format is ISO 8601 duration format
+        if periodDuration.hasPrefix("P") {
```

Comment:
Just an idea: we could use `ISOPeriodFormatter` to convert into `ISODuration` and then, from it, convert it into a `SubscriptionPeriod`. It wouldn't be so clean because we would need to check something like this: 

```
if isoDuration.years > 0 {
    return SubscriptionPeriod(value: isoDuration.years, unit: .year)
} else if isoDuration.months > 0 {
    return SubscriptionPeriod(value: isoDuration.months, unit: .month)
...
```
and so on. I think it could be better because we would be centralizing the parsing of this kind of period strings. FFTI though, as it's not perfect either